tests/Controllers/LoginTest.php

```diff
@@ -109,6 +111,15 @@ public function testAfterLoggedInNotDesplayLoginPage(): void
 
     public function testLoginActionUsernameSuccess(): void
     {
+        // Change the validation rules
+        $config           = new class () extends Validation {
+            public $login = [
+                'username' => 'required|max_length[30]|alpha_numeric_space|min_length[3]',
+                'password' => 'required',
+            ];
+        };
+        Factories::injectMock('config', 'Validation', $config);
```

Comment:
This is fine but since the Config Factories helper method returns the shared instance you can also just do this:
```suggestion
        config('Validation')->login = [
            'username' => 'required|max_length[30]|alpha_numeric_space|min_length[3]',
            'password' => 'required',
        ];
```

I realize after typing this that this doesn't work anymore once dynamic properties are deprecated 🤦‍♂️